src/stdlib_experimental_io.md

```diff
@@ -0,0 +1,120 @@
+# IO
+
+## Implemented
+
+ * `loadtxt`
+ * `open`
+ * `savetxt`
+
+
+## `loadtxt` - load a 2D array from a text file
+
+### Description
+Loads a rank-2 `array` from a text file.
+
+### Syntax
+
+`call loadtxt(filename, array)`
+
+### Arguments
+
+`filename`: Shall be  a character expression containing the file name from which to load the rank-2 `array`.
+
+`array`: Shall be an allocatable rank-2 array of type `real` or `integer`.
+
+### Return value
+
+Returns an allocated rank-2 `array` with the content of `filename`.
+
+### Example
+
+```fortran
+program test
+    use stdlib_experimental_io, only: loadtxt
+    implicit none
+    real, allocatable :: x(:,:)
+    call loadtxt('example.dat', x) 
+end program
+```
+
+
+## `open` - open a file
+
+### Description
+
+Returns the unit number of a file opened to read, to write, or to read and write. The file might be a text file or a binary file. All files are opened using a streamed access.
+
+### Syntax
+
+`u = open(filename [, mode] [, iostat])`
+
+### Arguments
+
+`filename`: Shall be a character expression containing the name of the file to open.
+
+`mode` (optional): Shall be a character expression containing characters describing the way in which the file will be used. The available modes are:
+
+
+| Character | Meaning |
+| --------- | ------- |
+| `r` | open for reading (default) |
+| `w` | open for writing, truncating the file first |
+| `x` | open for exclusive creation, failing if the file already exists |
+| `a` | open for writing, appending to the end of the file if it exists |
+| `b` | binary mode |
+| `t` | text mode (default) |
+| `+` | open for updating (reading and writing) |
```

Comment:
Maybe an extra sentence would be helpful to indicate which of these can be used together? I had to look at the code, to know I can select one letter r, w, x, a; one from b,t; and optionally the +. But perhaps this is not yet so critical as long as we are in the experimental stage.

Reply:
Thank you. I think it is important that we get a correct spec at this stage already. Otherwise it will lead to other PRs.

Would the following sentence be enough:

> `mode` may include one character to define which mode you want to open the file in (i.e., `r`, `w`, `x`, and `a`) and one character to specify if the file should be handled as a binary or a text file (i.e., `b`, or `t`). `mode` may optionally include a third character `+` to open the file for updating. The default `mode` is `rt` (i.e. open for reading a text file).

Reply:
I will reply later, however before I forget I think the characters in the table should be enclosed with quotation marks (single or double?) so they get rendered in Markdown as a character literals?
| `r` | open for reading (default) |
| `'r'` | open for reading (default) | <-- like this

The same follow for the modes mentioned in the text, e.g. The default `mode` is `'rt'`.

Reply:
@ivan-pi I modified the spec following your comments. Please let me know if it is now clear.

Reply:
Quotes should be inside the inline code block, not outside. So, `'r'`, not '`r`' as it is now.

Reply:
> Quotes should be inside the inline code block, not outside. So, `'r'`, not '`r`' as it is now.

Good. And modified.

Reply:
Compared to the Python documentation (e.g. [open](https://docs.python.org/3/library/functions.html#open)), the current explanation is relatively short and does not cover precisely what are the meanings of binary and text format and how they interact with the platform. 

But I do not want to hinder progress, so good to go from my side too. 👍